gf-webapp-admin/src/component/rooms/room/RoomItem.tsx

```diff
@@ -42,15 +44,79 @@ export const RoomItem = (props: RoomProps) => {
         <h2 className={styles.title}>
           {`Room #${room.roomNumber}`}
         </h2>
-        <RoomsType />
-        <Description />
-        <LongDescription />
-        <Services />
+        <SelectField
+          itemsList={Object.values(RoomType)}
+          selectedItem={room.type}
+          titleText='Room type'
+          onChangeValue={(value: string) => {
+            isValidRoomType(value) ?
+              room.type = value
+              :
+              null;
+            setRoom(room);
+          }}
+        />
+        <TextField
+          value={room.description}
+          titleText='Room description'
+          onChangeValue={(value: string) => {
```

Comment:
Many the similar functions onChangeValue. If possible better to move out and use one function for every component

Reply:
Textfield and other fields is a components with simple logic. `onChangeValue` works with room class. We can't put a room class in a text field and work with it inside

Reply:
Ok.